src/app/icd/IcdManagementServer.cpp

```diff
@@ -63,7 +73,8 @@ Status IcdManagementServer::UnregisterClient(PersistentStorageDelegate & storage
     if (!is_admin)
     {
         VerifyOrReturnError(verificationKey.HasValue(), InteractionModel::Status::Failure);
-        VerifyOrReturnError(verificationKey.Value().data_equal(entry.key), InteractionModel::Status::Failure);
+        // TODO Specs issue #7519
+        // VerifyOrReturnError(verificationKey.Value().data_equal(entry.key), InteractionModel::Status::Failure);
```

Comment:
this is going to cause a bunch of tests to fail

Reply:
Test are run with admin privilege.